components/renku_data_services/crc/models.py

```diff
@@ -242,16 +274,23 @@ def update(self, **kwargs: Any) -> "ResourcePool":
     @classmethod
     def from_dict(cls, data: dict) -> "ResourcePool":
         """Create the model from a plain dictionary."""
+        cluster = None
         quota: Optional[Quota] = None
-        classes = []
+        classes: list[ResourceClass] = []
+
         if "quota" in data and isinstance(data["quota"], dict):
             quota = Quota.from_dict(data["quota"])
         elif "quota" in data and isinstance(data["quota"], Quota):
             quota = data["quota"]
+
         if "classes" in data and isinstance(data["classes"], set):
             classes = [ResourceClass.from_dict(c) if isinstance(c, dict) else c for c in list(data["classes"])]
         elif "classes" in data and isinstance(data["classes"], list):
             classes = [ResourceClass.from_dict(c) if isinstance(c, dict) else c for c in data["classes"]]
+
+        if "cluster" in data and isinstance(data["cluster"], Cluster):
+            cluster = Cluster.from_dict(data["cluster"]) if isinstance(data["cluster"], dict) else data["cluster"]
```

Comment:
```suggestion
        if "cluster" in data:
            cluster = Cluster.from_dict(data["cluster"]) if isinstance(data["cluster"], dict) else data["cluster"]
```

Reply:
Done, I also had to add the case to handle None.